lensed_hosts/generate_lensed_hosts_agn.py

```diff
@@ -0,0 +1,309 @@
+    phl  = lens_P['phl']                # position angle of the lens, degree
+    eshr = lens_P['gamma']              # external shear
+    eang = lens_P['phg']                # position angle of external shear
+    ekpa = 0.0                          # external convergence
+
+    #----------------------------------------------------------------------
+    ai1, ai2 = ole.alphas_sie(xlc1, xlc2, phl, ql, rle, le,
+                              eshr, eang, ekpa, xi1, xi2)
+
+    yi1 = xi1 - ai1
+    yi2 = xi2 - ai2
+    #----------------------------------------------------------------------------
+    
+    lensed_mag_b_u, lensed_mag_b_g, lensed_mag_b_r, lensed_mag_b_i, lensed_mag_b_z, lensed_mag_b_y, lensed_image_b = lensed_sersic_2d(xi1,xi2,yi1,yi2,srcP_b,lens_P)
+
+    os.makedirs(os.path.join(outdir,'agn_lensed_bulges'), exist_ok=True)
+
+    fits_limg_b = os.path.join(outdir,'agn_lensed_bulges/') + str(lens_P['UID_lens']) + "_" + str(rle)+ "_" + str(lensed_mag_b_u)+"_"+str(lensed_mag_b_g)+"_"+str(lensed_mag_b_r)+"_"+str(lensed_mag_b_i)+"_"+str(lensed_mag_b_z)+"_"+str(lensed_mag_b_y)+ "_bulge.fits" 
+    #fits_limg_b = os.path.join(outdir,'agn_lensed_bulges/') + str(lens_P['UID_lens']) + "_" + str(xlc1)+ "_" + str(xlc2)+"_"+str(vd)+"_"+str(zl)+"_"+str(zs)+"_"+str(rle)+"_"+str(ql)+"_"+str(le)+"_"+str(phl)+"_"+str(eshr)+"_"+str(eang)+ "_bulge.fits" 
+
+    pyfits.writeto(fits_limg_b, lensed_image_b.astype("float32"), overwrite=True)
+
+    #----------------------------------------------------------------------------
+
+    lensed_mag_d_u, lensed_mag_d_g, lensed_mag_d_r, lensed_mag_d_i, lensed_mag_d_z, lensed_mag_d_y, lensed_image_d = lensed_sersic_2d(xi1,xi2,yi1,yi2,srcP_d,lens_P)
+
+    os.makedirs(os.path.join(outdir,'agn_lensed_disks'), exist_ok=True)
+
+   # fits_limg_d = os.path.join(outdir,'agn_lensed_disks/') + str(lens_P['UID_lens']) + "_" +str(rle)+ "_" + str(lensed_mag_d_u)+"_" +str(lensed_mag_d_g)+"_"+str(lensed_mag_d_r)+"_"+str(lensed_mag_d_i)+"_"+str(lensed_mag_d_z)+"_"+str(lensed_mag_d_y)+ "_disk.fits" 
+    fits_limg_d = os.path.join(outdir,'agn_lensed_disks/') + str(lens_P['UID_lens']) + "_" +str(rle)+ "_" + str(lensed_mag_d_u)+"_" +str(lensed_mag_d_g)+"_"+str(lensed_mag_d_r)+"_"+str(lensed_mag_d_i)+"_"+str(lensed_mag_d_z)+"_"+str(lensed_mag_d_y)+ "_disk.fits" 
+ 
+
+    pyfits.writeto(fits_limg_d, lensed_image_d.astype("float32"), overwrite=True)
+
+    return 0
+
+
+if __name__ == '__main__':
+
+    dsx = 0.01  # pixel size per side, arcseconds
+    nnn = 1000  # number of pixels per side
+    xi1, xi2 = ole.make_r_coor(nnn, dsx)
+
+    hdulist, ahb = load_in_data_agn()
+
+    #hdulist is the list of lens parameters
+    #ahb is the list of galaxy bulge and disk parameters
+
+    message_row = 0
+    message_freq = 50
+    for i, row in ahb.iterrows():
+        if i >= message_row:
+            print ("working on system ", i , "of", max(ahb.index))
```

Comment:
It looks to me like you are generating multiple FITS files per system. I think you are making one per lensed image. But each FITS stamp has all the lensed images of the host galaxy on the same postage stamp, right? So we should only be generating one FITS stamp per lensed system?